src/Avalonia.Visuals/Media/Imaging/BitmapDecodeOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Avalonia.Visuals.Media.Imaging;
+
+namespace Avalonia.Media.Imaging
+{
+    public struct BitmapDecodeOptions
+    {
+        public PixelSize DecodePixelSize { get; set; }
```

Comment:
Bitmap still needs a `Resize` method. If one wants to have a full bitmap and 4 thumbnails of different sizes, one shouldn't have to load the bitmap 5 times

Reply:
To be clear `Resize` isn't mutating the current instance and instead a new instance with the requested size is returned that uses the existing pixel data as the source.

`Bitmap Resize(PixelSize size, BitmapInterpolationMode mode)` for example

Reply:
Might be clearer to do resizing via a ctor or static method to make clear it's not mutating? e.g. `new Bitmap(Bitmap existing, PixelSize size, BitmapInterpolationMode mode)`?

Reply:
Xamarin does it via a static `CreateScaledBitmap` method for reference https://docs.microsoft.com/en-us/dotnet/api/android.graphics.bitmap.createscaledbitmap?view=xamarin-android-sdk-9

Reply:
Adding a ctor that excepts an existing bitmap should be fine.

```
public Bitmap(Stream stream, BitmapDecodeOptions options)
public Bitmap(string fileName, BitmapDecodeOptions options)
public Bitmap(Bitmap bitmap, BitmapDecodeOptions options)
```

Reply:
And y it could be just a static method

Reply:
@Gillibald im not sure how it would work internally, once you create the `Bitmap` instance at least skia backend turns it into skimage... pixels are no longer available.

Reply:
https://docs.microsoft.com/de-de/dotnet/api/skiasharp.skimage.peekpixels?view=skiasharp-1.60.2#SkiaSharp_SKImage_PeekPixels

Reply:
Pixmap represents the actual pixel data

Reply:
@Gillibald was able to use PeekPixels for CreateScaledBitmap implementation.